src/io.c

```diff
@@ -1792,24 +1896,35 @@ static Int InitKernel (
     // GAP strings which hold the latest lines read from the streams  and the
     // name of the current input file. For HPC-GAP we don't need the cookies
     // anymore, since the data got moved to thread-local storage.
-    Int i;
-    for ( i = 0;  i < ARRAY_SIZE(STATE(InputFiles));  i++ ) {
-      Cookie[i][0] = 's';  Cookie[i][1] = 't';  Cookie[i][2] = 'r';
-      Cookie[i][3] = 'e';  Cookie[i][4] = 'a';  Cookie[i][5] = 'm';
-      Cookie[i][6] = ' ';  Cookie[i][7] = '0'+i;
-      Cookie[i][8] = '\0';
-      InitGlobalBag(&(STATE(InputFiles)[i].stream), &(Cookie[i][0]));
-
-      MoreCookie[i][0] = 's';  MoreCookie[i][1] = 'l';  MoreCookie[i][2] = 'i';
-      MoreCookie[i][3] = 'n';  MoreCookie[i][4] = 'e';  MoreCookie[i][5] = ' ';
-      MoreCookie[i][6] = ' ';  MoreCookie[i][7] = '0'+i;
-      MoreCookie[i][8] = '\0';
-      InitGlobalBag(&(STATE(InputFiles)[i].sline), &(MoreCookie[i][0]));
+    for (Int i = 0; i < MAX_OPEN_FILES; i++) {
+        Cookie[i][0] = 's';
+        Cookie[i][1] = 't';
+        Cookie[i][2] = 'r';
+        Cookie[i][3] = 'e';
+        Cookie[i][4] = 'a';
+        Cookie[i][5] = 'm';
+        Cookie[i][6] = ' ';
+        Cookie[i][7] = '0' + i;
+        Cookie[i][8] = '\0';
+        InitGlobalBag(&(InputFiles[i].stream), &(Cookie[i][0]));
```

Comment:
I find it strange that we do this for `InputFiles[i].stream` but not for `OutputFiles[i].stream`. Isn't this an (old) bug? Or am I missing some reason why this is not necessary?

Reply:
The best person to answer this would be @stevelinton

I believe this is a bug.

Reply:
I'm fairly sure it is a bug, although it would take a little work to construct an example that demonstrated it. I think you'd need to do something like call PrintTo on a stream and during the execution of the PrintTo, contrive to lose all references to the stream from the GAP workspace, and garbage collect.

Reply:
@fingolfin I guess the fix, if we want it is out of scope for this PR

Reply:
Thanks all for looking at this and confirming the problem. Of course Steve is right that it is extremely unlikely to be triggered (otherwise I am sure somebody would have noticed it years ago). I'll see if I can add a fix here.

Reply:
Do you want to fix this in this PR? If so, I won't merge yet.